### DISCUSSION TITLE
Where can I find more reading or learning material about gpgpu + rust

### DISCUSSION THREAD

**Discussion by nihalpasham** [UpsettingBoy/gpgpu-rs#15] [General]
Hi, 

I was wondering if there's a write-up or an introductory blog-post (or video) for this work . The vast majority of `wgpu` based projects are focused on graphics i.e. real-time rendering. As a complete newbie to gpgpu computing (mostly work on embedded systems), I've been meaning to tinker with `gpgpu computing + rust` and after looking around for a while, I realize there aren't many (rusty) options at the moment. 

So, just wanted to say, thank you for this. Apart from the experimental async(ness), this repo (at least to my mind) also feels like a very good `getting started with gpgpu + rust guide`. I'd like wish you the best for wherever you take this.

**Comment by jeronimosg** [comment]
Thank you for the kind words!

There is no written material about this work more than the docs and commit messages (there may be some proper prior GH discussions). But I think the examples in the repo are very self-contained and more-or-less easy to understand (I'm biased haha).

For an introduction to GPGPU, I would recommend the [CUDA documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#introduction). They use very CUDA specific terminology, but the core concepts are the same for any other GPGPU framework out there.

Hope this helps!
